shared/src/commonMain/kotlin/com/example/tasky/alarm/data/AlarmLocalDataSource.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.tasky.alarm.data
+
+import com.example.tasky.database.AppDatabase
+import com.example.tasky.database.database
+import com.example.tasky.database.model.AgendaAlarmEntity
+
+class AlarmLocalDataSource(
+    private val appDatabase: AppDatabase = database,
+) {
+    suspend fun getAgendaAlarm(agendaId: String) = appDatabase.agendaAlarmDao().get(agendaId)
+
+    suspend fun getAllAgendaAlarm() = appDatabase.agendaAlarmDao().getAll()
+
+    suspend fun upsertAgendaAlarm(entity: AgendaAlarmEntity) = appDatabase.agendaAlarmDao().upsert(entity)
+
+    suspend fun deleteAgendaAlarm(agendaId: String) =
+        appDatabase.agendaAlarmDao().delete(
+            AgendaAlarmEntity(agendaId, 0),
+        )
+
+    suspend fun deleteAgendaAlarm(requestCode: Int) = appDatabase.agendaAlarmDao().deleteByRequestCode(requestCode)
+
+    suspend fun deleteAllAgendaAlarm() = appDatabase.agendaAlarmDao().deleteAll()
```

Comment:
You could technically be using the following structure, which would be perfectly fine to use. This is especially true since this class would live in your `data` layer while the Repo interface lives in the `domain` layer of your app.

```
class ReminderRepository(
  private val api: ReminderApi,
  private val dao: ReminderDao
)
```

I have used `LocalDataSource` and `RemoteDataSource` interfaces a lot in the past because of Clean Architecture. But I've found them to be just another layer of abstraction in most cases that does not serve a real purpose in the projects that I used them in. Now, I will only create them when it is needed in a project. And when I just forward calls like here, then that does not bring a lot of value to my project 😄 

Again, using them is not wrong; this is just my view on it. Feel free to continue using them.